sql/builtin_functions.h

```diff
@@ -152,10 +153,10 @@ struct RegisterBuiltin {
 
                     bool constantArgs = true;
                     for (auto& arg : args) {
-                        constantArgs = constantArgs && arg.metadata.isConstant;
+                        constantArgs = constantArgs && arg.info->isConst();
                     }
 
-                    result.resultMetadata.isConstant = constantArgs && determinism == DETERMINISTIC;
+                    result.resultInfo = result.resultInfo->getConst(constantArgs && determinism == DETERMINISTIC);
```

Comment:
If I understand correctly, this is done here to prevent checking for constness in every single builtin?  Don't we have cases where a builtin function would be deterministic despite the non-constness of its arguments?